x/posts/internal/types/msgs.go

```diff
@@ -41,8 +53,24 @@ func (msg MsgCreatePost) ValidateBasic() sdk.Error {
 	if msg.Creator.Empty() {
 		return sdk.ErrInvalidAddress(fmt.Sprintf("Invalid creator address: %s", msg.Creator))
 	}
-	if len(msg.Message) == 0 {
-		return sdk.ErrUnknownRequest("Post message cannot be empty")
+
+	if len(strings.TrimSpace(msg.Message)) == 0 {
+		return sdk.ErrUnknownRequest("Post message cannot be empty nor blank")
+	}
+
+	if len(strings.TrimSpace(msg.Subspace)) == 0 {
+		return sdk.ErrUnknownRequest("Post subspace cannot be empty nor blank")
+	}
+
+	if len(msg.OptionalData) > 10 {
```

Comment:
Better store the limits in constants rather than hardcoding here.

Reply:
Done